sdk/go/pulumi/stash.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pulumi
+
+import (
+	"reflect"
+)
+
+// Stash manages a reference to a Pulumi stash resource.
+type Stash struct {
+	CustomResourceState
+
+	// Output is any value stored in the stash resource
+	Output AnyOutput `pulumi:"output"`
+
+	// Input is the value passed to the stash resource
```

Comment:
I think `Input` could use a little more explanation, ie it's always passed through , or even explain what we'd use this for (primarily in tf convert?).